feat: Add an example for dotnet

CaptainStandby opened the pull request:
This adds a basic example on how to make use of Ory Network in an ASP.NET application.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

alnr commented on dotnet-ory-network/src/ExampleApp/Properties/launchSettings.json:
```diff
@@ -0,0 +1,37 @@
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
+      "applicationUrl": "https://localhost:7092;http://localhost:5286",
```
Haven't seen port `7092` anywhere else 🤷

CaptainStandby replied:
This is generated via `dotnet new mvc` and is part of the starter template.

My idea was to keep the changes made to the template at a minimum so it can be used as a reference. Alternatively I could strip everything out, that is not relevant to the specific use case of integrating Ory Network.

Any preferences on your part @alnr @aeneasr ?

alnr commented on dotnet-ory-network/src/ExampleApp/wwwroot/js/site.js:
```diff
@@ -0,0 +1,4 @@
+﻿// Please see documentation at https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification
```
do we need this file?

alnr commented on dotnet-ory-network/.devcontainer/Dockerfile:
```diff
@@ -0,0 +1,2 @@
+FROM mcr.microsoft.com/devcontainers/dotnet:0-7.0
+RUN curl https://raw.githubusercontent.com/ory/meta/master/install.sh | bash -s -- -b /usr/local/bin/ ory
```
I've seen a couple of missing newlines at EOF. Would you mind configuring your editor to add them on save? 👍

alnr commented on dotnet-ory-network/.vscode/launch.json:
```diff
@@ -0,0 +1,35 @@
+  "version": "0.2.0",
+  "configurations": [
+    {
+      // Use IntelliSense to find out which attributes exist for C# debugging
```
These JSON comments might interfere with our formatting jobs.

alnr commented on dotnet-ory-network/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Overview
+
+Generated using the default `dotnet new mvc` command.
```
Is there a way to easily distinguish all the generated files? Could we move them all to one directory, maybe?

alnr reviewed (changes requested):
I'm guessing there's no convenient way around committing all the generated files, is there?

Good work!

CaptainStandby commented:
I have made the following changes:
- reworked the example, so it doesn't contain all those nasty generated files from `dotnet new mvc`
- fixed line endings and file encodings
- fixed license headers
- added docker support and e2e test
- added ci workflow to run the e2e test

I will have to take a look at why the `django-ory-cloud` workflow fails

alnr reviewed (approved):
Spectacular!

vinckr reviewed (approved):
awesome 🎉 
thanks for also fixing the failing test, a dream
